runners/spark/src/main/java/org/apache/beam/runners/spark/aggregators/metrics/WithNamedAggregatorsSupport.java

```diff
@@ -0,0 +1,132 @@
+import com.google.common.collect.Maps;
+import com.google.common.collect.Ordering;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricFilter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+
+import java.util.Map;
+import java.util.SortedMap;
+
+/**
+ * A {@link MetricRegistry} decorator-like* that supports {@link AggregatorMetric} by exposing
+ * the underlying * {@link org.apache.beam.runners.spark.aggregators.NamedAggregators}'
+ * aggregators as {@link Gauge}s.
+ * <p>
+ * *{@link MetricRegistry} is not an interface, so this is not a by-the-book decorator.
+ * That said, it delegates all metric related getters to the "decorated" instance.
+ * </p>
+ */
+public class WithNamedAggregatorsSupport extends MetricRegistry {
+
+  private MetricRegistry internalMetricRegistry;
+
+  private WithNamedAggregatorsSupport(final MetricRegistry internalMetricRegistry) {
+    this.internalMetricRegistry = internalMetricRegistry;
+  }
+
+  public static WithNamedAggregatorsSupport forRegistry(final MetricRegistry
+                                                            metricRegistry) {
+    return new WithNamedAggregatorsSupport(metricRegistry);
+  }
+
+  @Override
+  public SortedMap<String, Timer> getTimers(MetricFilter filter) {
+    return internalMetricRegistry.getTimers(filter);
+  }
+
+  @Override
+  public SortedMap<String, Meter> getMeters(MetricFilter filter) {
+    return internalMetricRegistry.getMeters(filter);
+  }
+
+  @Override
+  public SortedMap<String, Histogram> getHistograms(MetricFilter filter) {
+    return internalMetricRegistry.getHistograms(filter);
+  }
+
+  @Override
+  public SortedMap<String, Counter> getCounters(final MetricFilter filter) {
+    return internalMetricRegistry.getCounters(filter);
+  }
+
+  @Override
+  public SortedMap<String, Gauge> getGauges(final MetricFilter filter) {
+    return
+        new ImmutableSortedMap.Builder<String, Gauge>(
+            Ordering.from(String.CASE_INSENSITIVE_ORDER))
+            .putAll(internalMetricRegistry.getGauges(filter))
+            .putAll(extractGauges(internalMetricRegistry, filter))
+            .build();
+  }
+
+  private Map<String, Gauge> transformToGauges(final AggregatorMetric metric) {
+    return Maps.transformValues(metric.getNamedAggregators().renderAll(),
+        new Function<Object, Gauge>() {
+
+          @Override
+          public Gauge apply(final Object rawGaugeObj) {
+            return new Gauge<Double>() {
+
+              @Override
+              public Double getValue() {
+                // at the moment the metric's type is assumed to be
+                // compatible with Double. While far from perfect, it seems reasonable at
+                // this point in time
+                return Double.parseDouble(rawGaugeObj.toString());
```

Comment:
While the limitation is understandable, and in most cases I'd expect Aggregator values to be `Long` or `Integer` anyway, it's probably best to completely ignore an "unsupported Guage".